bundles/org.openhab.core.model.sitemap/src/org/openhab/core/model/sitemap/Sitemap.xtext

```diff
@@ -170,6 +171,11 @@ Number returns ecore::EBigDecimal:
 XState returns ecore::EString:
     INT | ID | STRING | FLOAT;
 
+terminal INPUT_HINT returns ecore::EString:
+            '"' ("text" | "number" | "date" | "time" | "datetime") '"' |
+            "'" ("text" | "number" | "date" | "time" | "datetime") "'";
```

Comment:
Can we also ask a third alternative without quotes ?
```
terminal INPUT_HINT returns ecore::EString:
            ('"' ("text" | "number" | "date" | "time" | "datetime") '"') |
            ("'" ("text" | "number" | "date" | "time" | "datetime") "'") |
            ("text" | "number" | "date" | "time" | "datetime");
```

Reply:
> Wouldn't it be more secure to add parenthesis?
> 
> ```
> terminal INPUT_HINT returns ecore::EString:
>             ('"' ("text" | "number" | "date" | "time" | "datetime") '"') |
>             ("'" ("text" | "number" | "date" | "time" | "datetime") "'");
> ```

Would make it clearer for the reader, but is not required. I will change it. This is how the default terminal STRING is defined:

```
terminal STRING:
    '"' ( '\\' . /* 'b'|'t'|'n'|'f'|'r'|'u'|'"'|"'"|'\\' */ | !('\\'|'"') )* '"' |
    "'" ( '\\' . /* 'b'|'t'|'n'|'f'|'r'|'u'|'"'|"'"|'\\' */ | !('\\'|"'") )* "'"
```

Reply:
> Can we also add a third alternative without quotes ?
> 
> ```
> terminal INPUT_HINT returns ecore::EString:
>             ('"' ("text" | "number" | "date" | "time" | "datetime") '"') |
>             ("'" ("text" | "number" | "date" | "time" | "datetime") "'") |
>             ("text" | "number" | "date" | "time" | "datetime");
> ```

I am reluctant to do this. The period parameter for the Chart widget that cannot take quotes around it caused a lot of trouble in the mainUI sitemap parser. The JSON representation has quotes around it. It required special handling in a few places, see https://github.com/openhab/openhab-webui/pull/1745 and following PR's https://github.com/openhab/openhab-webui/pull/1781 and https://github.com/openhab/openhab-webui/pull/1783. So I would like to avoid it.
I actually think it was bad practice to mix this in the grammar in the first place. Booleans should be true/false without quotes, numbers should be without quotes and any other parameter should have quotes. The item parameter is OK as an exception. But at this point, I don't think we should change that, just not introduce new confusing ones.

Reply:
Ok with your arguments.